Assets/Scripts/GameGrid.cs

```diff
@@ -191,25 +193,82 @@ public int FindMatches()
         {
             // TODO Check for Max, max -1, max - 2, etc            
 
+            // Get ALL matching icons in all rows + columns, then calculate score and delete
+            List<Tuple<int, int>> matchList = new List<Tuple<int, int>>();
+
             // Search rows for matches
             for (int row = 0; row < _grid.GetLength(0); row++)
             {
+                // For each row member, check if next tag is same to increase counter
+                // If next tag is null or different, check if counter is > 3 and add match coordinates to matchList
+                int matchCounter = 1;
                 for (int col = 0; col < _grid.GetLength(1); col++)
                 {
-                    if (_grid.GetLength(1) - col < 3)
+                    if (col + 1 == _grid.GetLength(1) || !_grid[row, col].CompareTag(_grid[row, col + 1].tag))
```

Comment:
Looks like the first part of the condition should be inversed. `col + 1 != _grid.GetLength(1)`

Reply:
The first part checks if it's at the end of the row, because when I had just the right side by itself it had an error because _grid[row, col +1] didn't exist

Reply:
Ah yeah I thought you were looking for the next being a match. Didn't see the `!`. The code could be a bit cleaner if you invert the if check, increment the counter and `continue;`, which goes to the next iteration. Then your `matchCounter` check doesn't have to be in an if block and less indenting is needed.

Reply:
I don't get it, why would that remove matchCounter from the if block?

Reply:
Looking at the structure of the following logic, there is a bunch of logic in the `if`, but only one line in the `else`. You could bring what the `else` does to the top of this section of code, and `continue` after `matchCounter++`.

And with inverting the first `if`, it would look like this.
```
for (...)
{
  if (col +1 != _grid.GetLength(1) && _grid[row,col].CompareTag(_grid[row,col+1].tag))
  {
   matchCounter++;
   continue;
  }
  
  if (matchCounter >= 3)
  {
     // Rest of code
  }
}
```

It removes the nested `if` blocks.

Reply:
Oops you wouldn't `continue`, took that out